kotlin_intro/src/main/kotlin/com/infinum/homework/repository/InFileCarCheckUpRepository.kt

```diff
@@ -0,0 +1,78 @@
+package com.infinum.homework.repository
+
+import com.infinum.homework.exception.CarCheckUpNotFoundException
+import com.infinum.homework.model.Car
+import com.infinum.homework.model.CarCheckUp
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.core.io.Resource
+import org.springframework.stereotype.Repository
+import java.io.FileOutputStream
+import java.time.LocalDateTime
+
+@Repository
+@Qualifier("inFileCarCheckUpRepository")
+@ConditionalOnProperty(name = ["repository.switch"], havingValue = "turned-off")
+class InFileCarCheckUpRepository(private val carCheckUpsFileResource: Resource) :
```

Comment:
Nastavno na komentar vezano uz resource, ovo ti onda samo postane:
```suggestion
class InFileCarCheckUpRepository(@Value("\${data-source.dbName}") private val carCheckUpsFileResource: Resource) :
```

Također onda kao property vrijednost za `data-source.dbName` u application.properties možeš iskoristit keyword `classpath` u springu koji će ti odmah pokazat na `resource` direktorij, pa to onda bude ovakva vrijednost na kraju: `classpath:car_checkups.txt`

Reply:
Ovo je super, samo u ovakvoj implementaciji sam morao rucno stvoriti car_checkups.txt buduci da ako ga nema odmah baca Exception kroz constructor. Pa mi onda ovaj ispod if za provjeru da li taj file postoji ili ne nema nekog smisla.
Mozda sam nesto propustio pa slobodno bacis neki trik kako da rijesim trigeranje tog Exceptiona ako taj file vec nije kreiran.

Reply:
U pravu si tu, tu sam ja propustio da file već mora bit kreiran kad se ide tom metodom, slobodno zadrži prethodnu implementaciju 👌